### ISSUE TITLE
URLs are all missing last character in final output

### ISSUE THREAD

**Issue by gotlougit** [gotlougit/knowyourselfindia#1]
There must be some error while correcting relative links in the output from the Wikipedia API, as 
```html
<a href="https://en.wikipedia.org/wiki/Kapildeo_Sing" title="Kapildeo Singh">Kapildeo Singh</a>
```

is the output that is being delivered to the user, whereas the URL should actually be https://en.wikipedia.org/wiki/Kapildeo_Singh

It seems that the regex should be checked to deliver the correct output.

**Comment by gotlougit** [comment]
This regex query *should* work (from ```crawler/cleanse.py```):
```python
results = re.findall(r'href="[A-Za-z0-9&?=$#%.-_:/;]*"', i)
```
Not sure what's going wrong with it....

**Comment by gotlougit** [comment]
Found the error: it was not the regex that was going wrong, but rather a small error in the creation of the new URL to be written.

<details>
<summary>
The incorrect line
</summary>
<code>
newr =  'href="https://en.wikipedia.org/' + r[7:-2] + '"'
</code>
</details>

<details>
<summary>
The correct line
</summary>
<code>
newr =  'href="https://en.wikipedia.org/' + r[7:-1] + '"'
</code>
</details>

This issue should be fixed in commit e1827f8
